client/app/components/Accordion/index.jsx

```diff
@@ -45,7 +45,11 @@ export const Accordion = ({
   const inputClassNames = () => `${dark ? css.dark : ''} ${large ? css.large : ''}`;
 
   return (
-    <div id={`${id}_accordion`} className={inputClassNames()}>
+    <div
+      id={`${id}_accordion`}
+      className={inputClassNames()}
+      data-testid={`accordion-${open ? 'open' : 'closed'}`}
```

Comment:
+1

Reply:
So I wasn't ecstatic with this, thanks for calling this out and offering suggestions. I tried the `.toBeVisible()` function which I thought was a good idea, but unfortunately it did not work..

While taking another look at the dom I saw we are using role button on the accordion and have an aria label for expanded. Using that I think we come to a better test and I can revert this change to accordion. 

https://testing-library.com/docs/dom-testing-library/api-queries#bytestid
I do want to mention that (per the doc above) it is better to use testids over css classes. Reason being that it's much more fragile than a testid. With that said, they are both attributes in the dom that the user is not aware of and the whole point of this library like you referenced earlier is to test close to the user's experience. 

I think using the role button for the accordion and getting the aria expanded state is a marked improve over what I have here. Thanks again for pointing this out.

Reply:
That's a great point. I think testing against `aria-expanded` makes sense 👍

`toBeVisible` would definitely be nice, but I haven't had a chance yet to understand why it isn't working. If you find any insights there, please lmk! I came across a thread suggesting it might have something to do with how styles are processed/injected (link towards the end of the [comment here](https://github.com/ifmeorg/ifme/pull/1891/files#r502610778))? But I'm not sure.

I went through the docs you linked:

>That said, they are _way_ better than querying based on DOM structure or styling css class names.

Avoiding querying by classname makes sense. On the other hand, checking whether a classname exists on a node that has been queried by 'role' (as a substitute for checking for visibility) feels less generic. It wouldn't be as fragile as querying by classname, even though it's still less ideal. It's like checking an attribute and is nearer to the implementation. I like your idea for validating the aria attribute!